engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -296,10 +280,10 @@ private void processAggregateEmailsByAggregationKey(EmailAggregationKey aggregat
 
                 emailSender.sendEmail(aggregation.getKey(), event, subject, body);
             }
-        } finally {
-            if (delete) {
-                emailAggregationRepository.purgeOldAggregation(aggregationKey, endTime);
-            }
```

Comment:
I suspect that running `purgeOldAggregation` from a `finally` block does not work if an exception is thrown during a preceding DB operation. Indeed, the transaction is already dead and interactions with the DB are no longer possible.